web/modules/custom/bnf/bnf_client/src/Hook/SubscriptionHooks.php

```diff
@@ -38,4 +43,47 @@ public function queueUpdate(EntityInterface $entity): void {
     }
   }
 
+  /**
+   * What to do when a subscription is deleted.
+   */
+  #[Hook('bnf_subscription_delete')]
+  public function subscriptionDelete(EntityInterface $entity): void {
+    /** @var \Drupal\bnf_client\Entity\Subscription $entity */
+    Assert::isInstanceOf($entity, Subscription::class);
+
+    $field_key = 'bnf_source_subscriptions';
+
+    // Finding all nodes that have been imported as part of this subscription.
+    $nodeStorage = $this->entityTypeManager->getStorage('node');
+    $query = $nodeStorage->getQuery();
+    $nids =
+      $query
+        ->condition($field_key, [$entity->id()], 'IN')
+        // We want all nodes, even if the user does not have access.
+        ->accessCheck(FALSE)
+        ->execute();
+
+    /** @var \Drupal\node\NodeInterface[] $nodes */
+    $nodes = $nodeStorage->loadMultiple($nids);
+
+    // Looping through the nodes, and remove the reference to this subscription.
+    foreach ($nodes as $node) {
```

Comment:
This could potentially be quite a few nodes. Are we sure we'll be able to handle them all in `hook_delete`?